From 78ec356e59b77d1eb1f15e8d75336819353a5f7f Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Wed, 06 Mar 2024 14:44:07 +0200
Subject: [PATCH] [FSW-402] fs1: bootrom: add die type and sku type

with this patch, die type & sku type is added to
identify master die, slave io die and compute
die. sku type to identify single or multi die
sku. here after integer number comparison to
identify die and sku can be avoided.

Change-Id: I02050aa4cc6f63941892f980baa9ed1c66c497ea
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 585da0c..1aa63c6 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -14,6 +14,7 @@
 	.efuse_read = 0,
 	.curr_img = 0,
 	.die_id = 0,
+	.die_type = INVALID_DIE_TYPE,
 };
 
 struct boot_rom_manifest_s __attribute__((section (".rom_manifest_section")))boot_rom_manifest;
diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 691442c..3098ac5 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -17,3 +17,15 @@
 	return ((die_cpu_id & ARC_LOCAL_CONF_DIE_CPU_ID_DIE_ID_M)
 		>> ARC_LOCAL_CONF_DIE_CPU_ID_DIE_ID_S);
 }
+
+enum boot_rom_die_type boot_rom_get_die_type(void)
+{
+	if(boot_rom_global.die_id <= COMPUTE_DIE_3)
+		return COMPUTE_DIE;
+	else if(boot_rom_global.die_id == IO_DIE_0)
+		return IO_DIE_MASTER;
+	else if(boot_rom_global.die_id == IO_DIE_1)
+		return IO_DIE_SLAVE;
+	else
+		return INVALID_DIE_TYPE;
+}
\ No newline at end of file
diff --git a/include/boot_rom.h b/include/boot_rom.h
index 0880b85..bb06f9a 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -29,6 +29,31 @@
 /* extracts "n" number of bits from position "p" in the given number "v" */
 #define EXTRACT_BITS(v, p, n)	(((v) & (((1 << (n)) - 1) << (p))) >> (p))
 
+enum boot_rom_sku_type {
+	MULTI_DIE_SKU = 0,
+	SINGLE_DIE_SKU = 1,
+	INVALID_SKU = 2
+};
+
+enum boot_rom_die_id {
+	COMPUTE_DIE_0,
+	COMPUTE_DIE_1,
+	COMPUTE_DIE_2,
+	COMPUTE_DIE_3,
+	IO_DIE_0,
+	IO_DIE_0_HALF_1,
+	IO_DIE_1,
+	IO_DIE_1_HALF_1,
+	INVALID_DIE_ID,
+};
+
+enum boot_rom_die_type {
+	COMPUTE_DIE,
+	IO_DIE_MASTER,
+	IO_DIE_SLAVE,
+	INVALID_DIE_TYPE,
+};
+
 enum rom_internal_boot_desc_idx {
 	ROM_DESC_PPBOOT,
 	ROM_DESC_FAKE_FUSE,
@@ -49,6 +74,7 @@
 	uint8_t efuse_read;
 	uint8_t curr_img;
 	uint8_t die_id;
+	enum boot_rom_die_type die_type;
 	uint64_t ppboot_start_addr;
 };
 
@@ -61,6 +87,7 @@
 extern struct boot_rom_manifest_s boot_rom_manifest;
 
 uint8_t boot_rom_get_die_id(void);
+enum boot_rom_die_type boot_rom_get_die_type(void);
 void boot_rom_init(void);
 
 #endif /* _BOOT_ROM_H_ */
